lib/oli_web/controllers/api/directed_discussion_controller.ex

```diff
@@ -0,0 +1,120 @@
+# API to support getting collaborative space discussions into the directed discussion activity.
+
+defmodule OliWeb.Api.DirectedDiscussionController do
+  @moduledoc """
+  Provides user state service endpoints for extrinsic state.
+  """
+  use OliWeb, :controller
+  use OpenApiSpex.Controller
+
+  alias Oli.Repo
+  alias Oli.Delivery.Sections
+
+  alias Oli.Resources.Collaboration
+  alias Phoenix.PubSub
+  alias Oli.Resources.Collaboration.Post
+
+  def create_post(conn, %{"resource_id" => resource_id, "section_slug" => section_slug}) do
+    content = conn.body_params["content"]
+    parent_post_id = Map.get(conn.body_params, "parent_post_id", nil)
+    current_user = Map.get(conn.assigns, :current_user)
+    section = Sections.get_section_by_slug(section_slug)
+
+    Collaboration.create_post(%{
+      :status => :approved,
+      :user_id => current_user.id,
+      :section_id => section.id,
+      :resource_id => resource_id,
+      :parent_post_id => parent_post_id,
+      :thread_root_id => parent_post_id,
+      :replies_count => 0,
+      :anonymous => false,
+      :content => %{"message" => content}
+    })
```

Comment:
There is a shortcut syntax for maps with atom keys that can be used which looks like:

```
%{
  status: :approved,
  user_id: current_user.id,
  ...
}
```